src/proof/dot/dot_printer.h

```diff
@@ -19,13 +19,50 @@
 #define CVC5__PROOF__DOT__DOT_PRINTER_H
 
 #include <iostream>
+#include <stack>
 
 #include "printer/let_binding.h"
 #include "proof/proof_node.h"
 
 namespace cvc5::internal {
 namespace proof {
 
+/**
+ * An enumeration for nodes clusterS type. Each type defines in which cluster
+ * the node will be inserted when printed.
+ */
+enum class NodeClusterType : uint8_t
+{
+  // ======== First Scope
+  // Type of node cluster that is always in the root of the graph.
+  // The rule is always SCOPE.
+  FIRST_SCOPE = 0,
+  // ======== SAT
+  // Type of node cluster that is between First Scope and CNF.
+  // The rules are: CHAIN_RESOLUTION, FACTORING and REORDERING.
+  SAT,
+  // ======== CNF
+  // Type of node cluster that is below SAT and above THEORY_LEMMA or
+  // PRE_PROCESSING.
+  // The rules are in the range between NOT_NOT_ELIM and CNF_ITE_NEG3.
+  CNF,
+  // ======== THEORY_LEMMA
+  // Type of node cluster that is composed only by SCOPE nodes and stays in the
```

Comment:
Rather say that these are proof nodes that are contained in a scope starting just after the SAT +CNF proof (i.e. outside of preprocessing).